Many unknown_pragma warnings output under GCC

<!-- For bug report -->
# Environment

* OS version: Linux
* Elona foobar verision: `develop`


# Details
When compiling under GCC, there are many warnings of the following type.

```
/home/ruin/build/elonafoobar/src/ui/../data/types/../../lua_env/../shared_id.hpp:7: warning: ignoring #pragma clang diagnostic [-Wunknown-pragmas]
 #pragma clang diagnostic push
```

# Solution
Add `-Wunknown-pragmas` to the list of ignored warnings.

--- a/src/defines.hpp
+++ b/src/defines.hpp
@@ -18,6 +18,12 @@
 #endif
 
 
+// Compiler
+#if BOOST_COMP_CLANG
+#define ELONA_COMPILER_CLANG
+#endif
+
+
 // Endianness
 #if BOOST_ENDIAN_BIG_BYTE
 #define ELONA_BIG_ENDIAN

--- a/src/shared_id.hpp
+++ b/src/shared_id.hpp
@@ -1,15 +1,23 @@
 #pragma once
+
 #include <string>
 
-// Assumes that a compiler just ignores unknown pragmas.
-// Supresses "unused variable" warning due to Boost.Flyweight library.
-// This warning is reported only by Clang.
+#include "defines.hpp"
+
+// Supresses "unused variable" warning due to Boost.Flyweight library in Clang
+// compiler.
+#ifdef ELONA_COMPILER_CLANG
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wunused-variable"
+#endif
+
 #include <boost/flyweight.hpp>
 #include <boost/flyweight/no_locking.hpp>
 #include <boost/flyweight/no_tracking.hpp>
+
+#ifdef ELONA_COMPILER_CLANG
 #pragma clang diagnostic pop
+#endif
 
 
 

